c_check assumes nonempty $cross_suffix if run in a path with a dash

When executing make in a path with a dash it in, the `c_check` program automatically detects cross_suffix, but it seems like anything with a dash is assumed to be a cross compile. I don't know whether this is intentional or not, but perhaps a README warning is in order.
```
make[1]: /idi/yesimon-ar: Command not found
../Makefile.tail:40: recipe for target 'libs' failed
make[1]: *** [libs] Error 127
make[1]: Leaving directory '/idi/yesimon-scratch/yesimon/tmp/openblas20150724-14846-n6dn4n/OpenBLAS-0.2.14/interface'
Makefile:141: recipe for target 'libs' failed
make: *** [libs] Error 1
```
From `c_check`: https://github.com/xianyi/OpenBLAS/blob/3f1b57668e3d73944981d7bda194ca133f4660b7/c_check#L28

--- a/c_check
+++ b/c_check
@@ -1,5 +1,7 @@
 #!/usr/bin/perl
 
+use File::Basename;
+
 # Checking cross compile
 $hostos   = `uname -s | sed -e s/\-.*//`;    chop($hostos);
 $hostarch = `uname -m | sed -e s/i.86/x86/`;chop($hostarch);
@@ -26,14 +28,12 @@ if ($?) {
 
 $cross_suffix = "";
 
-if ($ARGV[0] =~ /(.*)(-[.\d]+)/) {
-    if ($1 =~ /(.*-)(.*)/) {
-	$cross_suffix = $1;
-    }
-} else {
-    if ($ARGV[0] =~ /([^\/]*-)([^\/]*$)/) {
-	$cross_suffix = $1;
-    }
+if (dirname($compiler_name) ne ".") {
+    $cross_suffix .= dirname($compiler_name) . "/";
+}
+
+if (basename($compiler_name) =~ /(.*-)(.*)/) {
+    $cross_suffix .= $1;
 }
 
 $compiler = "";

--- a/c_check
+++ b/c_check
@@ -243,7 +243,7 @@ print MAKEFILE "BINARY64=\n" if $binformat ne bin64;
 print MAKEFILE "BINARY32=1\n" if $binformat eq bin32;
 print MAKEFILE "BINARY64=1\n" if $binformat eq bin64;
 print MAKEFILE "FU=$need_fu\n" if $need_fu ne "";
-print MAKEFILE "CROSS_SUFFIX=$cross_suffix\n" if $cross_suffix ne "";
+print MAKEFILE "CROSS_SUFFIX=$cross_suffix\n" if $cross != 0 && $cross_suffix ne "";
 print MAKEFILE "CROSS=1\n" if $cross != 0;
 print MAKEFILE "CEXTRALIB=$linker_L $linker_l $linker_a\n";
 

